src/oc/storage/representations/entry.clj

```diff
@@ -50,49 +76,93 @@
 
   ([entry] entry))
 
-(defn- entry-and-links
+(defn- include-secure-uuid
+  "Include secure UUID property for authors."
+  [entry secure-uuid access-level]
+  (if (= access-level :author)
+    (assoc entry :secure-uuid secure-uuid)
+    entry))
+
+(defun- entry-and-links
   "
   Given an entry and all the metadata about it, render an access level appropriate rendition of the entry
   for use in an API response.
   "
-  [entry board-slug org-slug comments reactions access-level user-id]
+  ([entry board-slug org :guard map? comments reactions access-level user-id secure-access?]
   (let [entry-uuid (:uuid entry)
+        secure-uuid (:secure-uuid entry)
         org-uuid (:org-uuid entry)
+        org-slug (:slug org)
         board-uuid (:board-uuid entry)
+        draft? (= :draft (keyword (:status entry)))
         reactions (if (= access-level :public)
                     []
                     (content/reactions-and-links org-uuid board-uuid entry-uuid reactions user-id))
-        links [(self-link org-slug board-slug entry-uuid)
-               (up-link org-slug board-slug)]
-        full-links (cond 
-                    (= access-level :author)
+        links (if secure-access?
+                ;; secure UUID access
+                [(secure-self-link org-slug secure-uuid)]
+                ;; normal access
+                [(self-link org-slug board-slug entry-uuid)
+                 (up-link org-slug board-slug)])
+        more-links (cond 
+                    ;; Accessing drafts, or access by authors get editing links                    
+                    (or (and draft? (not secure-access?)) (= access-level :author))
```

Comment:
Why or the user is author or he's accessing drafts? to access drafts the user has to be an author, shouldn't the second or condition be enough?

Reply:
It's cause draft is an artificial board @bago2k4  so no one really has `author` access to  drafts in the traditional sense cause it doesn't really exist. So we check if they are on their drafts (which they are always on author of) OR they are an author of the board.

Hopefully that makes.